data/completion/snap

```diff
@@ -32,8 +32,14 @@ _complete() {
         try)
             _filedir -d
             ;;
+        connect)
```

Comment:
%% deletes the longest match of the `*:` suffix in the variable, so everything if it matches, so if the pattern matches `*:` (no plug or slot known) the whole thing is deleted and -z goes true, so do not add the space when completing.

This is indeed unnecessarily magic and unreadable. It would be a lot more clear as simply:
```
if [[ "$COMPREPLY" == *: ]]; then
```
No need for comments then, because we can actually read it.